File: lib_base/src/main/java/com/android/base/utils/common/Checker.java
package com.android.base.utils.common;

import java.util.Collection;
import java.util.Map;

/**
 * 对象检查工具
 */
public class Checker {

    private Checker() {

    }

    public static boolean isEmpty(Collection<?> data) {
        return data == null || data.size() == 0;
    }

    public static boolean notEmpty(Collection<?> data) {
        return !isEmpty(data);
    }

    public static boolean isNull(Object o) {
        return o == null;
    }

    public static boolean isEmpty(Map<?, ?> map) {
        return map == null || map.size() == 0;
    }

    public static boolean notEmpty(Map<?, ?> map) {
        return !isEmpty(map);
    }

    public static <T> boolean isEmpty(T[] t) {
        return t == null || t.length == 0;
    }

    public static <T> boolean notEmpty(T[] t) {
        return !isEmpty(t);
    }

    public static <T> T requireNonNull(T obj) {
        if (obj == null)
            throw new NullPointerException();
        return obj;
    }

    public static <T> T requireNonNull(T obj, String message) {
        if (obj == null)
            throw new NullPointerException(message);
        return obj;
    }

    public static boolean nonNull(Object obj) {
        return obj != null;
    }

}

File: lib_base/src/main/java/com/android/base/utils/common/Checker.kt
@file:JvmName("Checker")

package com.android.base.utils.common

fun isEmpty(data: Collection<*>?): Boolean {
    return data == null || data.isEmpty()
}

fun notEmpty(data: Collection<*>?): Boolean {
    return !isEmpty(data)
}

fun isNull(o: Any?): Boolean {
    return o == null
}

fun isEmpty(map: Map<*, *>?): Boolean {
    return map == null || map.isEmpty()
}

fun notEmpty(map: Map<*, *>): Boolean {
    return !isEmpty(map)
}

fun <T> isEmpty(t: Array<T>?): Boolean {
    return t == null || t.isEmpty()
}

fun <T> notEmpty(t: Array<T>): Boolean {
    return !isEmpty(t)
}

fun <T> requireNonNull(obj: T?): T {
    if (obj == null) {
        throw NullPointerException("requireNonNull failed")
    }
    return obj
}

fun <T> requireNonNull(obj: T?, message: String): T {
    if (obj == null) {
        throw NullPointerException("requireNonNull failed: %s".format(message))
    }
    return obj
}

fun nonNull(obj: Any?): Boolean {
    return obj != null
}
